auth/base.go

```diff
@@ -22,15 +22,29 @@ package auth
 import (
 	"errors"
 	"fmt"
+	"net/http"
+	"reflect"
+
 	"github.com/garyburd/redigo/redis"
 	"github.com/julienschmidt/httprouter"
 	"github.com/mittwald/servicegateway/config"
-	logging "github.com/op/go-logging"
-	"net/http"
+	"github.com/op/go-logging"
 )
 
 var InvalidCredentialsError error = errors.New("invalid credentials given")
 
+type AuthenticationIncompleteError struct {
+	AdditionalProperties map[string]interface{}
+}
+
+func (e AuthenticationIncompleteError) Error() string {
+	return fmt.Sprint("authentication incomplete")
+}
+
+func (e AuthenticationIncompleteError) Is(target error) bool {
+	return reflect.TypeOf(target) == reflect.TypeOf(e)
+}
```

Comment:
The standard libraries `error` module has a function for exactly this: https://pkg.go.dev/errors#Is

EDIT: Also, apparently you're already using `errors.Is` further down, which would make this just dead code.

Reply:
The documentation says "An error is considered to match a target if it is equal to that target or if it implements a method Is(error) bool such that Is(target) returns true". So the seemingly unneccessary Is Method is called by `errors.Is`, if given. 

Further, correct me if i'm wrong, but an equality check on structs returns true if the structs are equal in terms of their fields values, otherwise, returns false. In this szenario we hardly can know, which exact properties the AuthenticationIncompleteError holds, so we cannot really check for equality, but have to compare the types, which I tried at this point.

Reply:
Oh, I must've missed that part, but you're right. Nevermind, then. 🤐